### PR TITLE
Added coinchange implementation

### PR DESCRIPTION
**Pull request by nursyah21**

### CODE REVIEW

File: dynamic/coinchange.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package dynamic
+
+// return combination of coin
```
Comment (raklaptudirm) [inline dynamic/coinchange.go:3]: Make this comment to be more thorough, and start with the function name.
```go
// CoinChange ...
// ...
```

File: dynamic/coinchange.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+func CoinChange(coins []int32, amount int32) int32 {
+	combination := make([]int32, amount)
+	combination[0] = 1
+
+	//Time Complexity O(mn)
```
Comment (raklaptudirm) [inline dynamic/coinchange.go:8]: These comments are unnecessary.

**Review by raklaptudirm — APPROVED** [review]
Looks good to me.

File: dynamic/coinchange.go (hunk 2 above)
Comment (nursyah21) [reply dynamic/coinchange.go:8]: can I change the code directly here, without having to open a pull request again?.

Comment (raklaptudirm) [reply dynamic/coinchange.go:8]: Yeah of course.

File: dynamic/coinchange.go [hunk 3]
```diff
@@ -1,15 +1,12 @@
 package dynamic
 
-// return combination of coin
+// CoinChange
```
Comment (raklaptudirm) [inline dynamic/coinchange.go:3]: I think you are not getting the idea of go doc comments. A few examples may help:
https://github.com/TheAlgorithms/Go/blob/master/math/prime/millerrabinprimalitytest.go#L99
https://github.com/TheAlgorithms/Go/blob/master/math/prime/millerrabinprimalitytest.go#L119

Comment (nursyah21) [reply dynamic/coinchange.go:3]: Can you give me suggestion, what should i write. i still can't understand

is this correct?
```
// CoinChange find number of combination coins to get amount target
```

Comment (raklaptudirm) [reply dynamic/coinchange.go:3]: I think the language can be improved upon.

Comment (nursyah21) [reply dynamic/coinchange.go:3]: i only can think to write like this, make a simple comment, i look another example they make comment like this too just simple and describe what does the function do
```
// CoinChange find number of combination coins to get amount target
```

Comment (raklaptudirm) [reply dynamic/coinchange.go:3]:
```go
// CoinChange finds the number of possible combinations of coins
// of different values which can get to the target amount.
```

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Please remove the changes made to the `go.mod` and the `go.sum` files.

**Comment by nursyah21** [comment]
> Please remove the changes made to the `go.mod` and the `go.sum` files.
how do to remove changes commmit

**Comment by raklaptudirm** [comment]
Just edit those files.

**Comment by raklaptudirm** [comment]
Why did you close the pr?

**Pull request closed without merging**